src/tiledb/cloud/config.py

```diff
@@ -11,10 +11,20 @@
 
 default_host = "https://api.tiledb.com"
 
-config = configuration.Configuration()
+_config = configuration.Configuration()
 default_config_file = Path.joinpath(Path.home(), ".tiledb", "cloud.json")
 
 
+def __getattr__(name):
+    global logged_in
+    if name == "config":
+        if not logged_in:
+            logged_in = load_configuration(default_config_file)
+        return _config
+    else:
+        raise AttributeError
```

Comment:
This keeps the cloud.json file from being loaded until code accesses `config.config`. Like at line 727: https://github.com/TileDB-Inc/TileDB-Cloud-Py/pull/681/files#diff-a6634d675335a1b2739c0e82ea86301bb69c1b37e408be28a629002d54644711R727.

Reply:
Does this also work correctly if the user does `from tiledb.cloud.config import config`?

Reply:
@ihnorton Yes:

```
>>> from tiledb.cloud.config import config
/Users/sean.gillies/projects/TileDB-Inc/TileDB-Cloud-Py/.venv/lib/python3.9/site-packages/tiledb/cloud/config.py:103: UserWarning: You must first login before you can run commands. Please run tiledb.cloud.login.
  warnings.warn(
>>> config.api_key
{}
```
```
>>> import tiledb.cloud                                                                                                                                                                                              
>>> tiledb.cloud.login(token="******")
>>> config.api_key
{'X-TILEDB-REST-API-KEY': '******'}
```

Reply:
Success ✅